### PR TITLE
Gaussian PES Scanning

### PR DESCRIPTION
**Pull request by shivupa**
This PR adds the unrelaxed scan gaussian logfile and updates the scan tests for unrestricted and restricted scans and updates the docs. Let me know if anything needs to be changed.

### CODE REVIEW

**Review by langner — COMMENTED** [review]
Thanks! One main question: do you think we should strive to have both relaxed and unrelaxed scans in our unit test logfiles, or just one and the other in regressions? I would lean towards the latter.

File: cclib/parser/gaussianparser.py [hunk 1]
```diff
@@ -1002,11 +1010,15 @@ def extract(self, inputfile, line):
...
             hyphens = next(inputfile)
             line = next(inputfile)
+            scanparm = [[] for _ in range(len(self.scannames))]
             while line != hyphens:
                 broken = line.split()
                 self.append_attribute('scanenergies', (utils.convertor(float(broken[-1]), "hartree", "eV")))
-                self.append_attribute('scanparm', [float(p) for p in broken[1:-1]])
+                for idx,p in enumerate(broken[1:-1]):
+                    scanparm[idx].append(float(p))
+                #self.append_attribute('scanparm', [float(p) for p in broken[1:-1]])
```
Comment (langner) [inline cclib/parser/gaussianparser.py:1019]: Clean this up?

**Comment by shivupa** [comment]
> Thanks! One main question: do you think we should strive to have both relaxed and unrelaxed scans in our unit test logfiles, or just one and the other in regressions? I would lean towards the latter.

The only reservation I would have is that the inputs and outputs of relaxed and unrelaxed scans are different so we have different parts of the code handling both (at least in the gaussian parser and this will be the case in the orca parser).

**Comment by shivupa** [comment]
the regression tests `2D-PES-all-converged.log` and probably `2D-PES-one-unconverged.log` will fail. The eigenvalues line only has stars. I can take a closer look later

**Comment by langner** [comment]
That's fine, I'm going to merge this in and we can figure out the kinks later.

**Pull request merged**